From 57d01f39039611232cbf55405b088687ee185eb4 Mon Sep 17 00:00:00 2001
From: Muthukumar K <muthukumar.kandasamy@intel.com>
Date: Thu, 30 Aug 2012 20:31:13 +0530
Subject: Tombstone while doing video streaming when server sends non-standard 90kHz (SRC assertion failure)

BZ: 50674

Feng server sets 90k sample rate for all the streams (both
audio and video). The CHECK here results in a crash as 90k
is not the standard one. The sample rate is recalculated in
the codec with the stream data. Thus removal of this CHECK
does no harm.

Category: AOSP Improvement
Domain: AUDIO_MEDIA-Audio Framework
Origin: Internal
Upstream-Candidate: no, Properietary

Orig-Change-Id: I1c9e703071d2b125fa3f5fde30b7d7baccc1ff62

Change-Id: I9162b18e5d477ae106e41f0efb041c52fe279a8d
Signed-off-by: Muthukumar K <muthukumar.kandasamy@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 .../rtsp/AMPEG4ElementaryAssembler.cpp             |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/rtsp/AMPEG4ElementaryAssembler.cpp b/media/libstagefright/rtsp/AMPEG4ElementaryAssembler.cpp
index 2ace475..ad58c80 100644
--- a/media/libstagefright/rtsp/AMPEG4ElementaryAssembler.cpp
+++ b/media/libstagefright/rtsp/AMPEG4ElementaryAssembler.cpp
@@ -103,6 +103,7 @@ static bool GetSampleRateIndex(int32_t sampleRate, size_t *tableIndex) {
         }
     }
 
+    ALOGW("%d is a non-standarad sample rate.", sampleRate);
     return false;
 }
 
@@ -192,7 +193,9 @@ AMPEG4ElementaryAssembler::AMPEG4ElementaryAssembler(
                 desc.c_str(), &sampleRate, &numChannels);
 
         mChannelConfig = numChannels;
-        CHECK(GetSampleRateIndex(sampleRate, &mSampleRateIndex));
+        // CHECK removed here as SR is recalculated in codec from stream data.
+        // Feng server sets 90k SR default. This removal is necessary then.
+        GetSampleRateIndex(sampleRate, &mSampleRateIndex);
     }
 }
 
-- 
1.7.4.1

